mri/numerics/reconstruct.py

```diff
@@ -443,7 +443,7 @@ def sparse_rec_pogm(gradient_op, linear_op, prox_op, cost_op=None,
         print("Done.")
         print("Execution time: ", end - start, " seconds")
         print("-" * 40)
-    x_final = opt.x_final
+    x_final = linear_op.adj_op(opt.x_final)
```

Comment:
This is done to follow the same norm as other optimizers. We want to ensure that we return Images in the end.

Reply:
I really dislike this as the `adj_op` is not necessarily an inverse (or reconstruction) operation. I don't know how it's done elsewhere, but we should let the user change the outcome the way they want.

Reply:
For a more general perspective, the sparse_rec_* is supposed to give the image solution to the reconstruction problem, if we assume that this is the goal of the function than we should output the image. This is for a general perspective, I think that to handle this in a correct manner, `linar_op` should have both an adjoint operator and an inverse operator, i.e. for the simplest case -- i.e. orthogonal wavelet transform, handling correctly the boundaries --  the adjoint and the inverse should be the same whereas for more complicated  transforms maybe we should implement a dedicated inverse operation.

Reply:
I agree with Loubna, resolving here. We could instance an issue to obtain the reconstruction operator in both WaveletN and WaveletUD classes.

Reply:
Yes sorry I was misreading, I thought this was the algorithm. This is def fine.